crates/accelerate/src/sparse_observable/mod.rs

```diff
@@ -2941,6 +3354,118 @@ impl PySparseObservable {
+    /// operator that "performs ``a``, and then performs ``b`` on the result".  The argument
+    /// ``front=True`` instead makes this a right multiplication.
+    ///
+    /// ``self`` and ``other`` must be the same size, unless ``qargs`` is given, in which case
+    /// ``other`` can be smaller than ``self``, provided the number of qubits in ``other`` and the
+    /// length of ``qargs`` match.  ``qargs`` can never contain duplicates or indices of qubits that
+    /// do not exist in ``self``.
+    ///
+    /// Beware that this function can cause exponential explosion of the memory usage of the
+    /// observable, as the alphabet of :class:`SparseObservable` is not closed under composition;
+    /// the composition of two single-bit terms can be a sum, which multiplies the total number of
+    /// terms.  This memory usage is not _necessarily_ inherent to the resultant observable, but
+    /// finding an efficient re-factorization of the sum is generally equally computationally hard.
+    /// It's better to use domain knowledge of your observables to minimize the number of terms that
+    /// ever exist, rather than trying to simplify them after the fact.
+    ///
+    /// Args:
+    ///     other: the observable used to left-multiply ``self``.
+    ///     qargs: if given, the qubits in ``self`` to associated with the qubits in ``other``.  Put
+    ///         another way: if this is given, it is similar to a more efficient implementation of::
+    ///
+    ///             self.compose(other.apply_layout(qargs, self.num_qubits))
+    ///
+    ///         as no temporary observable is created to store the applied-layout form of ``other``.
+    ///     front: if ``True``, then right-multiply by ``other`` instead of left-multiplying
+    ///         (default ``False``).  The ``qargs`` are still applied to ``other``.  This is most
+    ///         useful when ``qargs`` is set, or ``other`` might be an object that must be coerced
+    ///         to :class:`SparseObservable`.
+    #[pyo3(signature = (other, /, qargs=None, *, front=false))]
+    fn compose<'py>(
+        &self,
+        other: &Bound<'py, PyAny>,
+        qargs: Option<&Bound<'py, PyAny>>,
+        front: bool,
+    ) -> PyResult<Bound<'py, PySparseObservable>> {
+        let py = other.py();
+        let Some(other) = coerce_to_observable(other)? else {
+            return Err(PyTypeError::new_err(format!(
+                "unknown type for compose: {}",
+                other.get_type().repr()?
+            )));
+        };
+        let other = other.borrow();
+        let inner = self.inner.read().map_err(|_| InnerReadError)?;
+        let other_inner = other.inner.read().map_err(|_| InnerReadError)?;
+        if let Some(order) = qargs {
+            if other_inner.num_qubits() > inner.num_qubits() {
+                return Err(PyValueError::new_err(format!(
+                    "argument ({}) has more qubits than the base ({})",
```

Comment:
Would this be clearer to read? 
```suggestion
                    "argument has more qubits ({}) than the base ({})",
```

Reply:
Sure, if you prefer.  Done in 5dd0666c99.